### PR TITLE
Code optimizations reported by static code analysis (2018-11-12) 

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Performance "Code optimizations reported by static code analysis"

#### Purpose of change

Cleanup code while increase performance and fix minor error.

#### Describe the solution

See commit messages.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
Travis builds are failed due to tests which is unrelated to current changes.

File: src/debug.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #endif
 
 #ifdef TILES
-#include <SDL.h>
+#include "sdl_wrappers.h"
```
Comment (BevapDin) [inline src/debug.cpp]: Why that? "sdl_wrappers.h" includes more than just "SDL.h" and this file does not use any of the additional stuff declared in "sdl_wrappers.h" only. So it basically just includes more files without any advantage.

File: src/crash.cpp [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include <typeinfo>
 
 #ifdef TILES
-#include <SDL.h>
+#include "sdl_wrappers.h"
```
Comment (BevapDin) [inline src/crash.cpp]: Why that? "sdl_wrappers.h" includes more than just "SDL.h" and this file does not use any of the additional stuff declared in "sdl_wrappers.h" only. So it basically just includes more files without any advantage.

File: src/item.cpp [hunk 3]
```diff
@@ -1533,13 +1533,13 @@ std::string item::info( std::vector<iteminfo> &info, const iteminfo_query *parts
...
 
-        temp1.str( "" );
+        temp1.clear();
         temp1 << _( "Used on: " ) << enumerate_as_string( mod.usable.begin(),
         mod.usable.end(), []( const gun_type_type & used_on ) {
             return string_format( "<info>%s</info>", used_on.name().c_str() );
         } );
 
-        temp2.str( "" );
+        temp1.clear();
```
Comment (Qrox) [inline src/item.cpp]: Should be `temp2.clear();`

File: src/sdl_wrappers.cpp [hunk 4]
```diff
@@ -4,10 +4,6 @@
 
 #include "debug.h"
 
-#include <SDL.h>
-#include <SDL_ttf.h>
-#include <SDL_image.h>
```
[the remark this answers is not in the document]
Comment (ZhilkinSerg) [reply src/sdl_wrappers.cpp]: Thanks.

**Pull request merged**